lib/upgrade/etcd/conversionv1v3/hostendpoint_test.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+	"github.com/projectcalico/libcalico-go/lib/numorstring"
+)
+
+var hepTable = []struct {
+	description string
+	v1API       unversioned.Resource
+	v1KVP       *model.KVPair
+	v3API       apiv3.HostEndpoint
+}{
+	{
+		description: "Valid basic v1 hep has data moved to right place",
+		v1API: apiv1.HostEndpoint{
+			Metadata: apiv1.HostEndpointMetadata{
+				Name: "my-hep",
+				Node: "my-node",
+			},
+			Spec: apiv1.HostEndpointSpec{
+				InterfaceName: "eth3",
+			},
+		},
+		v1KVP: &model.KVPair{
+			Key: model.HostEndpointKey{
+				EndpointID: "my-hep",
+				Hostname:   "my-node",
+			},
+			Value: &model.HostEndpoint{
+				Name: "eth3",
+			},
+		},
+		v3API: apiv3.HostEndpoint{
+			ObjectMeta: v1.ObjectMeta{
+				Name: "my-node.my-hep",
+			},
+			Spec: apiv3.HostEndpointSpec{
+				InterfaceName: "eth3",
+				Node:          "my-node",
+			},
+		},
+	},
+	{
+		description: "Valid filled v1 hep has data moved to right place",
+		v1API: apiv1.HostEndpoint{
+			Metadata: apiv1.HostEndpointMetadata{
+				Name: "my-hep",
+				Node: "my-node",
+			},
+			Spec: apiv1.HostEndpointSpec{
+				ExpectedIPs:   []cnet.IP{
+					cnet.MustParseIP("192.168.1.1"),
+					cnet.MustParseIP("fe80::"),
+				},
+				Profiles:      []string{"my-prof"},
+				InterfaceName: "eth3",
+				Ports: []apiv1.EndpointPort{
+					{
```

Comment:
Could you add a couple more ports:
- One with only a protocol
- One with only a Port number

Reply:
Are these just supposed to check that we don't lose the data when only one field is specified?